### PR TITLE
NonGeneric collections cleanup

### PR DESCRIPTION
**Pull request by Maxwe11**

### CODE REVIEW

**Comment by YingP99** [comment]
@Maxwe11 thanks for your work. Are you doing these by hand or using tool - such as VS 2014?

**Comment by krwq** [comment]
Looks good to me! :+1:

**Comment by joshfree** [comment]
Hi @maxwe11,
Thanks for your contribution. Just a note that ArrayList, SortedList, and Hashtable unit tests haven't been open sourced on github just yet -- but they're coming soon. I'd prefer to hold off on taking changes to ArrayList and Hashtable until after the unit tests come online in another week or so.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 1]
```diff
@@ -71,10 +69,7 @@ public ArrayList(int capacity)
             if (capacity < 0) throw new ArgumentOutOfRangeException("capacity", SR.Format(SR.ArgumentOutOfRange_MustBeNonNegNum, "capacity"));
             Contract.EndContractBlock();
 
-            if (capacity == 0)
-                _items = Array.Empty<Object>();
-            else
-                _items = new Object[capacity];
+            _items = capacity == 0 ? Array.Empty<Object>() : new Object[capacity];
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Why make this change? I think it is less readable in this form and also unnecessarily churns the code.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 2]
```diff
@@ -320,17 +315,18 @@ public virtual bool Contains(Object item)
...
             {
                 for (int i = 0; i < _size; i++)
+                {
                     if (_items[i] == null)
                         return true;
+                }
                 return false;
             }
-            else
+            for (int i = 0; i < _size; i++)
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: We should avoid removing these else clauses while technically correct right now, it unnecessarily churns the code and sets us up for potential issues later. For example right now all the places in that if block return but someone could end up adding a branch that doesn't return and then this code would be incorrect and cause a subtle bug.

Comment (Maxwe11) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Ok, I'll revert such changes

**Comment by Maxwe11** [comment]
@joshfree :ok_hand:

**Comment by Maxwe11** [comment]
What if replace KeyValuePairs class (see a80e1cd) with `KeyValuePair<object, object>` ?

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 3]
```diff
@@ -1048,17 +1044,18 @@ public override int IndexOf(Object value, int startIndex, int count)
...
                 {
                     for (int i = startIndex; i < endIndex; i++)
+                    {
                         if (_list[i] == null)
                             return i;
+                    }
                     return -1;
                 }
-                else
+                for (int i = startIndex; i < endIndex; i++)
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Please leave the else.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 4]
```diff
@@ -1120,17 +1117,18 @@ public override int LastIndexOf(Object value, int startIndex, int count)
...
                 {
                     for (int i = startIndex; i >= endIndex; i--)
+                    {
                         if (_list[i] == null)
                             return i;
+                    }
                     return -1;
                 }
-                else
+                for (int i = startIndex; i >= endIndex; i--)
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Please leave the else.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 5]
```diff
@@ -2459,11 +2457,7 @@ public bool MoveNext()
                     _currentElement = _list[++_index];
                     return true;
                 }
-                else
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Please leave the else.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs [hunk 6]
```diff
@@ -2472,8 +2466,10 @@ public Object Current
                 get
                 {
                     if (_index < _startIndex)
+                    {
                         throw new InvalidOperationException(SR.InvalidOperation_EnumNotStarted);
-                    else if (_index > _endIndex)
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Please leave the else.

File: src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs (hunk 2 above)
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs]: Thanks for understanding, I will assume you revert all the changes where you removed else statements.

File: src/System.Collections.NonGeneric/src/System/Collections/CollectionBase.cs [hunk 7]
```diff
@@ -33,12 +32,7 @@ protected CollectionBase(int capacity)
 
         protected ArrayList InnerList
         {
-            get
-            {
-                if (_list == null)
-                    _list = new ArrayList();
-                return _list;
-            }
+            get { return _list ?? (_list = new ArrayList()); }
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/CollectionBase.cs]: Please leave revert changes like this as well. I don't think they add enough value to actually warrant the change.

File: src/System.Collections.NonGeneric/src/System/Collections/DictionaryBase.cs [hunk 8]
```diff
@@ -22,12 +20,7 @@ public abstract class DictionaryBase : IDictionary
...
         protected Hashtable InnerHashtable
         {
-            get
-            {
-                if (_hashtable == null)
-                    _hashtable = new Hashtable();
-                return _hashtable;
-            }
+            get { return _hashtable ?? (_hashtable = new Hashtable()); }
         }
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/DictionaryBase.cs]: Please revert.

File: src/System.Collections.NonGeneric/src/System/Collections/Hashtable.cs [hunk 9]
```diff
@@ -736,11 +733,7 @@ protected virtual bool KeyEquals(Object item, Object key)
         // 
         public virtual ICollection Keys
         {
-            get
-            {
-                if (_keys == null) _keys = new KeyCollection(this);
-                return _keys;
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/Hashtable.cs]: Please revert this change but do feel free to put this if block on its own line to make it match the coding guidelines.

File: src/System.Collections.NonGeneric/src/System/Collections/Hashtable.cs [hunk 10]
```diff
@@ -755,11 +748,7 @@ public virtual ICollection Keys
         // 
         public virtual ICollection Values
         {
-            get
-            {
-                if (_values == null) _values = new ValueCollection(this);
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/Hashtable.cs]: Please revert this change but do feel free to put this if block on its own line to make it match the coding guidelines.

File: src/System.Collections.NonGeneric/src/System/Collections/KeyValuePairs.cs [hunk 11]
```diff
@@ -14,14 +14,14 @@
 
 namespace System.Collections
 {
-    [DebuggerDisplay("{value}", Name = "[{key}]", Type = "")]
+    [DebuggerDisplay("{Value}", Name = "[{Key}]", Type = "")]
     internal class KeyValuePairs
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/KeyValuePairs.cs]: Thanks for fixing this as it was truly a bug. I would however change this to be the field names _value and _key as that will be cheaper to execute in the debugger. While this is a simple property it still requires a function call in the Expression Evaluator which is more expensive than a field look-up.

File: src/System.Collections.NonGeneric/src/System/Collections/ReadOnlyCollectionBase.cs [hunk 12]
```diff
@@ -22,12 +20,7 @@ public abstract class ReadOnlyCollectionBase : ICollection
 
         protected ArrayList InnerList
         {
-            get
-            {
-                if (_list == null)
-                    _list = new ArrayList();
-                return _list;
-            }
+            get { return _list ?? (_list = new ArrayList()); }
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/ReadOnlyCollectionBase.cs]: Please revert.

File: src/System.Collections.NonGeneric/src/System/Collections/SortedList.cs [hunk 13]
```diff
@@ -455,8 +454,7 @@ public virtual Object GetKey(int index)
         // 
         public virtual IList GetKeyList()
         {
-            if (_keyList == null) _keyList = new KeyList(this);
-            return _keyList;
+            return _keyList ?? (_keyList = new KeyList(this));
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/SortedList.cs]: Please revert this change but do feel free to put this if block on its own line to make it match the coding guidelines.

File: src/System.Collections.NonGeneric/src/System/Collections/SortedList.cs [hunk 14]
```diff
@@ -472,8 +470,7 @@ public virtual IList GetKeyList()
         // 
         public virtual IList GetValueList()
         {
-            if (_valueList == null) _valueList = new ValueList(this);
-            return _valueList;
+            return _valueList ?? (_valueList = new ValueList(this));
```
Comment (weshaggard) [inline src/System.Collections.NonGeneric/src/System/Collections/SortedList.cs]: Please revert this change but do feel free to put this if block on its own line to make it match the coding guidelines.

**Pull request closed without merging**